gcm_filters/filter.py

```diff
@@ -201,32 +171,22 @@ def filter_func(field, *args):
         field_bar = laplacian.prepare(field_bar)
 
         for n in range(filter_spec.n_iterations):
-            for i in range(filter_spec.n_steps_total):
-                if filter_spec.is_laplacian[i]:
-                    s_l = np.real(filter_spec.s[i])
-                    tendency = laplacian(field_bar)  # Compute Laplacian
-                    if not Laplacian.is_dimensional:
-                        tendency = _maybe_dimensionalize(
-                            tendency, filter_spec.dx_min_sq
-                        )  # dimensionalize
-                    field_bar += (1 / s_l) * tendency  # Update filtered field
-                else:
-                    s_b = filter_spec.s[i]
-                    temp_l = laplacian(field_bar)  # Compute Laplacian
-                    temp_b = laplacian(
-                        temp_l
-                    )  # Compute Biharmonic (apply Laplacian twice)
-                    if not Laplacian.is_dimensional:
-                        temp_l = _maybe_dimensionalize(
-                            temp_l, filter_spec.dx_min_sq
-                        )  # dimensionalize
-                        temp_b = _maybe_dimensionalize(
-                            temp_b, filter_spec.dx_min_sq**2
-                        )  # dimensionalize
-                    field_bar += (
-                        temp_l * 2 * np.real(s_b) / np.abs(s_b) ** 2
-                        + temp_b * 1 / np.abs(s_b) ** 2
+            T_minus_2 = field_bar.copy()
```

Comment:
I'm responding here for all of the `.copy()` changes. I'm afraid to remove these because Python might just have `T_minus_2` and `field_bar` point to the same thing, such that every time we update `T_minus_2` it also updates `field_bar`, which we don't want.